src/calico_manifests.py

```diff
@@ -0,0 +1,157 @@
+"""This module provides the CalicoManifests class for managing Calico manifests."""
+import json
+import logging
+from base64 import b64encode
+from typing import Dict
+
+from charms.kubernetes_libs.v0.etcd import EtcdReactiveRequires
+from ops.manifests import ConfigRegistry, Manifests, Patch
+
+log = logging.getLogger(__name__)
+
+
+class PatchCalicoConflist(Patch):
+    """A patch class for modifying the Calico CNI Conflist in a ConfigMap."""
+
+    def __call__(self, obj) -> None:
+        """Modify the Calico Conflist within the given ConfigMap object."""
+        if not (obj.kind == "ConfigMap" and obj.metadata.name == "calico-config"):
+            return
+
+        log.info("Patching Calico Conflist.")
+        data = obj.data
+        if not data:
+            log.warning("calico-config: Unable to patch conflist, data not found.")
+            return
+
+        json_config = data.get("cni_network_config")
+
+        if not json_config:
+            return
+
+        conflist = json.loads(json_config)
+
+        for plugin in conflist.get("plugins"):
+            if plugin.get("type") == "calico":
+                mtu = self.manifests.config.get("mtu")
+                plugin["mtu"] = mtu if mtu else 0
```

Comment:
I think patching the mtu in the conflist becomes unnecessary if you restore the `__CNI_MTU__` value in the manifest and instead patch the `veth_mtu` value in the ConfigMap. See my other comments about those.

Reply:
Thanks! Removed 👍🏻